design/UsageValidationRules.md

```diff
@@ -62,3 +75,9 @@ If this is deemed too costly, we could only have two sub-resources tracked in te
 This would mean for example that a texture couldn't be used as both  `STORAGE` and a read-only usage inside a render pass.
 The state tracking required in implementation would become significantly simpler at the cost of flexibility for the applications.
 When using the more constrained version of usage validation for textures, the cost of validation is O(commands).
+
+### Opt-in for cross-dispatch data-races
+
+Enforcing no data-races between dispatches could potentially results in a lot of ALU time being wasted on big GPUs.
+We could allow applications to opt into cross-dispatch data-races to get some of the performance back by manually placing "UAV barriers".
```

Comment:
I haven't heard an argument on why this is preferred over just having separate compute passes. Perhaps, we could omit the part about `manualStorageBarriers` completely from this PR until there is more of a consensus? Or do you believe that it's essential in order to be able to say that there can be data races between dispatches?

Reply:
As discussed in yesterday's call I believe cross-dispatch will be essential for some applications. The consensus is that implementations take care of synchronization between dispatches so we can't make synchronization happen only at pass boundary, at least not without opt-in.